library/ssl_cli.c

```diff
@@ -2554,11 +2554,11 @@ static int ssl_parse_server_ecdh_params_psa( mbedtls_ssl_context *ssl,
      */
 
     if( end - *p < 4 )
-        return( MBEDTLS_ERR_SSL_BAD_HS_SERVER_KEY_EXCHANGE );
+        return( MBEDTLS_ERR_SSL_DECODE_ERROR );
 
     /* First byte is curve_type; only named_curve is handled */
     if( *(*p)++ != MBEDTLS_ECP_TLS_NAMED_CURVE )
-        return( MBEDTLS_ERR_SSL_BAD_HS_SERVER_KEY_EXCHANGE );
+        return( MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER );
```

Comment:
Yes, I agree.

Reply:
Hm, actually we don't know at this point: It could be valid but unsupported, or invalid.

Reply:
The ideal-world solution would be to check all legal values, and return `HANDSHAKE_FAILURE` for those that are unsupported, and return `ILLEGAL_PARAMETER` for all that are invalid. 

But this is impractical because it would bloat the code for no benefit except a more expressive error handling.

Reply:
`MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE` seems to be a better choice in such a situation.